src/python_testing/TestConformanceTest.py

```diff
@@ -0,0 +1,136 @@
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from typing import Any
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+from spec_parsing_support import build_xml_clusters, build_xml_device_types
+from TC_DeviceConformance import DeviceConformanceTests
+
+
+def create_onoff_endpoint(endpoint: int) -> dict[int, dict[int, dict[int, Any]]]:
+    # Really simple device with one endpoint that includes scenes management, which is provisional
+    # I'm ONLY populating the global attributes since the conformance test only uses these.
+    endpoint_tlv = {endpoint: {}}
+
+    on_off_device_type_id = 0x0100
+    on_off_device_type_revision = 3
+
+    # Descriptor
+    attr = Clusters.Descriptor.Attributes
+    attrs = {}
+    attrs[attr.FeatureMap.attribute_id] = 0
+    attrs[attr.AcceptedCommandList.attribute_id] = []
+    attrs[attr.GeneratedCommandList.attribute_id] = []
+    attrs[attr.ClusterRevision.attribute_id] = 3
+    attrs[attr.DeviceTypeList.attribute_id] = [Clusters.Descriptor.Structs.DeviceTypeStruct(
+        deviceType=on_off_device_type_id, revision=on_off_device_type_revision)]
+    attrs[attr.ServerList.attribute_id] = [Clusters.Identify.id,
+                                           Clusters.Groups.id, Clusters.ScenesManagement.id, Clusters.OnOff.id]
+    attrs[attr.ClientList.attribute_id] = []
+    attrs[attr.PartsList.attribute_id] = []
+    attrs[attr.AttributeList.attribute_id] = []
+    attrs[attr.AttributeList.attribute_id] = list[attrs.keys()]
+
+    endpoint_tlv[endpoint][Clusters.Descriptor.id] = attrs
+
+    # Identify
+    attr = Clusters.Identify.Attributes
+    attrs = {}
+    attrs[attr.FeatureMap.attribute_id] = 0
+    attrs[attr.AcceptedCommandList.attribute_id] = [Clusters.Identify.Commands.Identify.command_id]
+    attrs[attr.GeneratedCommandList.attribute_id] = []
+    attrs[attr.ClusterRevision.attribute_id] = 5
+    attrs[attr.IdentifyTime.attribute_id] = 0
+    attrs[attr.IdentifyType.attribute_id] = Clusters.Identify.Enums.IdentifyTypeEnum.kNone
+    attrs[attr.AttributeList.attribute_id] = []
+    attrs[attr.AttributeList.attribute_id] = list[attrs.keys()]
+
+    endpoint_tlv[endpoint][Clusters.Identify.id] = attrs
+
+    # OnOff
+    attr = Clusters.OnOff.Attributes
+    attrs = {}
+    # device type requires LT feature
+    attrs[attr.FeatureMap.attribute_id] = Clusters.OnOff.Bitmaps.Feature.kLighting
+    cmd = Clusters.OnOff.Commands
+    attrs[attr.AcceptedCommandList.attribute_id] = [cmd.Off.command_id, cmd.On.command_id, cmd.Toggle.command_id,
+                                                    cmd.OffWithEffect.command_id, cmd.OnWithRecallGlobalScene.command_id, cmd.OnWithTimedOff.command_id]
+    attrs[attr.GeneratedCommandList.attribute_id] = []
+    attrs[attr.ClusterRevision.attribute_id] = 6
```

Comment:
Would be good to have constants for these revision numbers, all at the top.

Reply:
moved up.